include/gridtools/stencil/frontend/expandable_run.hpp

```diff
@@ -147,8 +147,19 @@ namespace gridtools {
             }
 
             template <size_t Factor, class Comp, class Backend, class Grid, class... Fields, size_t... Is>
-            void run_impl(Comp comp, Backend, Grid const &grid, std::index_sequence<Is...>, Fields &&... fields) {
+            auto run_impl(Comp comp, Backend, Grid const &grid, std::index_sequence<Is...>, Fields &&... fields)
+                -> void_t<decltype(comp(make_arg<Is, Fields>()...))> {
                 using spec_t = decltype(comp(make_arg<Is, Fields>()...));
+                static_assert(meta::is_instantiation_of<frontend_impl_::spec, spec_t>::value,
+                    "Invalid stencil composition specification.");
```

Comment:
Is it possible to make this more precise? Even I wouldn't know what could have gone wrong.

Reply:
This assert triggers when the user passes a functor that returns some garbage (our void) instead of using our `EDSL` (`execute_parallel`, `multi_satge` etc).
Example:
  `run ([]{}, backend_t(), ...)`

Reply:
I think the wording is correct here. We can expand it to smth. like "the first argument is expected to be a functor that returns one of ours DSL stences: blal-bla ...". But it looks like overkill to me.

Reply:
I think you are right, stencil composition is a term the user should know.